7.23.0 blog post (#2826)
babel/website: website/blog/2023-09-22-7.23.0.md

---
layout: post
title:  "7.23.0 Released: Decorator Metadata and many new `import` features!"
author: Babel Team
authorURL: https://twitter.com/babeljs

> date:   2022-09-22 0:00:00

Author: ```suggestion
date:   2022-09-25 0:00:00
```

categories: announcements
share_text: "Babel 7.23.0 Released"
---

We just released Babel 7.23.0! 🎉

> It includes transform support for the [Decorator Metadata](https://github.com/tc39/proposal-decorator-metadata/), [Deferred Import Evaluation](https://github.com/tc39/proposal-defer-import-eval), and [Optional Chaining Assignment](https://github.com/tc39/proposal-optional-chaining-assignment) proposals, as well as parsing support for the [Source Phase Import](https://github.com/tc39/proposal-source-phase-imports) proposal. We also updated our parser to support the new TypeScript 5.2 version, and added a transform option to let you use the `.ts` extension in TypeScript imports.

Reviewer: ```suggestion
It includes transform support for the [Decorator Metadata](https://github.com/tc39/proposal-decorator-metadata/), [Deferred Import Evaluation](https://github.com/tc39/proposal-defer-import-eval), and [Optional Chaining Assignment](https://github.com/tc39/proposal-optional-chaining-assignment) proposals, as well as parsing support for the [Source Phase Import](https://github.com/tc39/proposal-source-phase-imports) proposal. We also updated our parser to support the new TypeScript 5.2 version, and added a transform option to let you use the `.ts` extension within TypeScript imports.
```

> Other than version 7.23.0, we recently published the first Babel 8 alpha release!

Reviewer: ```suggestion
In addition to releasing version 7.23.0, we also recently published the first Babel 8 alpha release!
```

```javascript
import source fooSource from "foo";
```

The motivating use case is to have static syntax to load `WebAssembly.Module` objects, rather than being forced to do so dynamically:

```javascript
// Before
const url = import.meta.resolve("./my-wasm/module.wasm");
const responsePromise = fetch(url);
const mod = await WebAssembly.compileStreaming(responsePromise);

// After
import source mod from "./my-wasm/module.wasm";
```

Babel now supports transforming source imports for WebAssembly modules targeting Node.js, Web-compatible environments (browsers and Deno), or both, depending on your configured [`targets`](https://babeljs.io/docs/options#targets).

You can transform it using the `@babel/plugin-proposal-import-wasm-source` plugin:

```json
{
  "targets": ["chrome 90", "firefox 90", "node 20.6"],
  "plugins": ["@babel/plugin-proposal-import-wasm-source"]
}
```

### Deferred Import Evaluation ([#15845](https://github.com/babel/babel/pull/15845), [#15878](https://github.com/babel/babel/pull/15878))

> The [Deferred Import Evaluation](https://github.com/tc39/proposal-defer-import-eval/) Stage 2 proposal allows deferring the (synchronous) evaluation of imported modules until when they are used. This can have greatly improve startup performance of your code, in case of modules with a significant initialization cost that aren't immediately used.

Reviewer: ```suggestion
The [Deferred Import Evaluation](https://github.com/tc39/proposal-defer-import-eval/) Stage 2 proposal allows deferring the (synchronous) evaluation of imported modules until they are used. This can greatly improve the startup performance of your code when using modules that have a significant initialization cost.
```
